frontend/src/services/profile_handler_service.ts

```diff
@@ -0,0 +1,54 @@
+import { PersonProps, MockProfileBackendService } from "./mock_profile_backend";
+
+/** Error that occurs when Id does not exist */
+export class NonExistentProfileError extends Error {
+  constructor(readonly id: string) {
+    super(`Profile ${id} does not exist`);
+  }
+}
+
+/** Error that occurs if Id was unable to be added */
+export class FailureToUpdateProfile extends Error {
+  constructor(readonly id: string) {
+    super(`Unable to update Profile ${id}`);
+  }
+}
+
+interface ProfileHandlerServiceAPI {
+  getPerson(id: string): Promise<PersonProps>;
+  updatePerson(PersonProps): Promise<boolean>;
+  deletePerson(id: string): Promise<boolean>;
+}
+
+/**
+ * Profile Handling service that manages list of user profiles using a mocked
+ * backend database.
+ */
+export class ProfileHandlerService implements ProfileHandlerServiceAPI {
```

Comment:
This service doesn't seem to really add any logic on top of backend service, so it might be worth rethinking things a bit. We could either remove this service and have the Profile component call BackendService directly, or we could move some logic out of BackendService and into this one.

The way I was thinking about it in my example is that BackendService would be a very thin wrapper around the actual API, so you'd give it some JSON and it would give you some JSON back. It would then be up to ProfileHandlerService to convert that JSON into something useful like a PersonProps object.

Reply:
I now have profile_handler_service taking in a person object, converting it to json, and handing it to the backend. Also, the backend now returns person json objects, which the ProfileServiceHandler will convert and send back to Profile.tsx. I think this addresses this comment, but if not I'd be more than happy to have a quick chat on how we can rethink it.